nixos/modules/services/security/crowdsec.nix

```diff
@@ -0,0 +1,959 @@
+
+    autoUpdateService = lib.mkEnableOption "if `true` `cscli hub update` will be executed daily. See `https://docs.crowdsec.net/docs/cscli/cscli_hub_update/` for more information";
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to automatically open firewall ports for `crowdsec`.
+      '';
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "The user to run crowdsec as";
+      default = "crowdsec";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = "The group to run crowdsec as";
+      default = "crowdsec";
+    };
+
+    name = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        Name of the machine when registering it at the central or local api.
+      '';
+      default = config.networking.hostName;
+      defaultText = lib.literalExpression "config.networking.hostName";
+    };
+
+    localConfig = lib.mkOption {
+      type = lib.types.submodule {
+        options = {
+          acquisitions = lib.mkOption {
+            type = lib.types.listOf format.type;
+            default = [ ];
+            description = ''
+              A list of acquisition specifications, which define the data sources you want to be parsed.
+              See <https://docs.crowdsec.net/docs/data_sources/intro> for details.
+            '';
+            example = [
+              {
+                source = "journalctl";
+                journalctl_filter = [ "_SYSTEMD_UNIT=sshd.service" ];
+                labels = {
+                  type = "syslog";
+                };
+              }
+            ];
+          };
+          scenarios = lib.mkOption {
+            type = lib.types.listOf format.type;
+            default = [ ];
+            description = ''
+              A list of scenarios specifications.
+              See <https://docs.crowdsec.net/docs/scenarios/intro> for details.
+            '';
+            example = [
+              {
+                type = "leaky";
+                name = "crowdsecurity/myservice-bf";
+                description = "Detect myservice bruteforce";
+                filter = "evt.Meta.log_type == 'myservice_failed_auth'";
+                leakspeed = "10s";
+                capacity = 5;
+                groupby = "evt.Meta.source_ip";
+              }
+            ];
+          };
+          parsers = lib.mkOption {
+            type = lib.types.submodule {
+              options = {
+                s00Raw = lib.mkOption {
+                  type = lib.types.listOf format.type;
+                  default = [ ];
+                  description = ''
+                    A list of stage s00-raw specifications. Most of the time, those are already included in the hub, but are presented here anyway.
+                    See <https://docs.crowdsec.net/docs/parsers/intro> for details.
```

Comment:
Q: is it intended that these "See xyz for details" notes are on the same line or a separate line to the rest of the description?

Having a line break here is great for reading the code, regardless of how you want the docs to render, and complies with the [one-sentence-per-line](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#one-sentence-per-line) guidance. However if you want to _render_ a line break or new paragraph then it's not enough:

- If you want to render a plain line break, you must end the prior line with a `\` character.
- If you want to render a paragraph break, you must separate the paragraphs with a blank line.

Reply:
I'll change that.

Reply:
I added a blank line to the descriptions where there's a "See xyz for details". I think this is fine as a new paragraph.